.gitignore

```diff
@@ -1 +1,8 @@
-bin
\ No newline at end of file
+bin
+*.hi
+*.o
+*.dyn_hi
+*.dyn_o
+*.swp
+*~
+\#*\#
```

Comment:
you might be interested in this https://stackoverflow.com/questions/151945/how-do-i-control-how-emacs-makes-backup-files